sdk/go/pulumi/rpc.go

```diff
@@ -457,32 +456,33 @@ func unmarshalPropertyValue(v resource.PropertyValue) (interface{}, bool, error)
 	case v.IsResourceReference():
 		ref := v.ResourceReferenceValue()
 
-		resName := ref.URN.Name()
+		resName := ref.URN.Name().String()
 		resType := ref.URN.Type()
 
 		var resource Resource
 		var err error
 
 		isProvider := tokens.Token(resType).HasModuleMember() && resType.Module() == "pulumi:providers"
 		if isProvider {
-			pkgName := resType.Name()
-			resourcePackageV, ok := resourcePackages.Load(packageKey(string(pkgName), ref.PackageVersion))
+			pkgName := resType.Name().String()
+			resourcePackageV, ok := resourcePackages.Load(pkgName)
```

Comment:
@pgavlin, I assume we want to support registrations of multiple versions like we do in the other languages, right?

If so, `resourcePackages` will need to be changed to be a map of `string` to `[]ResourcePackage` (array), rather than a single `ResourcePackage` value, e.g.:

```
-var resourcePackages sync.Map // map[string]ResourcePackage
+var resourcePackages sync.Map // map[string][]ResourcePackage
```

And the lookup will need to change to loop through the array to find a matching value based on the requested version.

And similar for `resourceModules`.

Since we're currently using `sync.Map` here, does this mean we'd need to move to add locking around the entire store and lookup operations (that would now deal with an array value)?

Reply:
Looks like this answer could be an option: https://stackoverflow.com/a/62527995

Reply:
Looking into it further, I wonder if it makes sense to ditch `sync.Map` entirely and just handle the concurrency with a normal `map[string][]ResourcePackage` or `map[string]map[string]ResourcePackage`

Reply:
> Looking into it further, I wonder if it makes sense to ditch sync.Map entirely and just handle the concurrency with a normal map[string][]ResourcePackage or map[string]map[string]ResourcePackage

A `map[string][]ResourcePackage` with manual synchronization sounds good to me.